app/models/agents/website_agent.rb

```diff
@@ -106,24 +106,41 @@ class WebsiteAgent < Agent
 
       Set `http_success_codes` to an array of status codes (e.g., `[404, 422]`) to treat HTTP response codes beyond 200 as successes.
 
+      The `interpolate` option is the way to format extracted data, and add meta information to each resulted payload.  Its value must be a hash, whose key-value pairs are interpolated after extraction and added to each event payload.  e.g.:
+
+          "interpolate": {
```

Comment:
If you need the header values, doesn't `PostAgent -> WebsiteAgent -> EventFormattingAgent` work?

The `WebsiteAgent` already does two very different jobs, fetching an URL and extracting HTML/JSON/text. Arbitrary Liquid interpolation as a third responsibility would further water down the scope of the Agent.

If your problem can not be solved with a combination of Agents we probably need to add the feature, but I think we should look into splitting up the `WebsiteAgent` into two or more Agents to prevent situations like this in the future.